cli/internal/run/run.go

```diff
@@ -253,18 +256,29 @@ func (r *run) run(ctx gocontext.Context, targets []string) error {
 		turboJSON.GlobalDeps,
 		pkgDepGraph.PackageManager,
 		pkgDepGraph.Lockfile,
+		turboJSON.GlobalPassthroughEnv,
+		r.opts.runOpts.EnvMode,
 		r.base.Logger,
 	)
 
 	if err != nil {
 		return fmt.Errorf("failed to collect global hash inputs: %v", err)
 	}
 
-	if globalHash, err := fs.HashObject(getGlobalHashable(globalHashable)); err == nil {
-		r.base.Logger.Debug("global hash", "value", globalHash)
-		g.GlobalHash = globalHash
+	if globalHashable.envMode == util.Infer && globalHashable.envVarPassthroughs == nil {
```

Comment:
Opt into the old global hash behavior.

Reply:
I'm not clear on whether we need this switch anymore. I think the reasoning was that since we're already busting global cache, we might as well drop the `getOldGlobalHashable`, but I think we can avoid busting global cache altogether, so we should keep the `getOldGlobalHashable` all around